services/wallet/bridge/swap_paraswap.go

```diff
@@ -0,0 +1,204 @@
+package bridge
+
+import (
+	"context"
+	"errors"
+	"math/big"
+	"strconv"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	ethTypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/status-im/status-go/account"
+	"github.com/status-im/status-go/eth-node/types"
+	"github.com/status-im/status-go/params"
+	"github.com/status-im/status-go/rpc"
+	walletCommon "github.com/status-im/status-go/services/wallet/common"
+	"github.com/status-im/status-go/services/wallet/thirdparty/paraswap"
+	"github.com/status-im/status-go/services/wallet/token"
+	walletToken "github.com/status-im/status-go/services/wallet/token"
+	"github.com/status-im/status-go/transactions"
+)
+
+var zeroAddress = common.Address{}
+
+type SwapTxArgs struct {
+	transactions.SendTxArgs
+	ChainID uint64 `json:"chainId"`
+}
+
+type SwapParaswap struct {
+	paraswapClient *paraswap.ClientV5
+	priceRoute     paraswap.Route
+	transactor     *transactions.Transactor
+}
+
+func NewSwapParaswap(rpcClient *rpc.Client, transactor *transactions.Transactor, tokenManager *walletToken.Manager) *SwapParaswap {
+	return &SwapParaswap{
+		paraswapClient: paraswap.NewClientV5(walletCommon.EthereumMainnet),
+		transactor:     transactor,
+	}
+}
+
+func (s *SwapParaswap) Name() string {
+	return "Paraswap"
+}
+
+func (s *SwapParaswap) Can(from, to *params.Network, token *walletToken.Token, toToken *walletToken.Token, balance *big.Int) (bool, error) {
+	if token == nil || toToken == nil {
+		return false, errors.New("token and toToken cannot be nil")
+	}
+
+	if from.ChainID != to.ChainID {
+		return false, nil
+	}
+
+	s.paraswapClient.SetChainID(from.ChainID)
+
+	searchForToken := token.Address == zeroAddress
+	searchForToToken := toToken.Address == zeroAddress
+	if searchForToToken || searchForToken {
+		tokensList, err := s.paraswapClient.FetchTokensList(context.Background())
```

Comment:
should this be cached for some time? i don't see that changing every minute.

I guess the client should do caching right?

Reply:
We don't do this always, only when the token has no address set, which should be no case or a rare case.